### PR TITLE
API Conformance report: C++ & Merge XML Part

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - **Summary** was separated to 3 parts: `Summary` (Just Interface), `OpSummary` (to create Conformance Operation report) and `ApiSummary` (to create `XML` file represented API plugin realization)
 - **APIBaseTest** was added as a interface to report API<->Behavior test results. `set_api_entity` is a virtual method to set property which will be validated by a test.
 - **APIConformance** scope was aligned according OpenVINO entities. Test prefixes were changed according structure.
 - **Test result counters overflow issue** was fixed (Incorrect result numbers)
 - **Namespace alignment.** A namespaces were changed: `ov::test::behavior` for OpenVINO2.0 API and `BehaviorTests` for old API 
 - **Python infrastructure** to merge xml files was changed to be able to merge different report types

### Tickets:
 - *CVS-66429*

###Example of report:
image

### CODE REVIEW

File: src/tests/functional/plugin/shared/include/base/ov_behavior_test_utils.hpp [hunk 1]
```diff
@@ -56,24 +85,21 @@ class OVInferRequestTests : public testing::WithParamInterface<InferRequestParam
...
+        std::tie(target_device, configuration) = this->GetParam();
+        function = ov::test::behavior::getDefaultNGraphFunctionForTheDevice(target_device);
         ov::AnyMap params;
         for (auto&& v : configuration) {
             params.emplace(v.first, v.second);
         }
-        execNet = core->compile_model(function, targetDevice, params);
-    }
-
-    void TearDown() override {
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/base/ov_behavior_test_utils.hpp:70]: to return

File: src/tests/functional/plugin/shared/include/behavior/infer_request/config.hpp [hunk 2]
```diff
@@ -18,56 +18,52 @@ typedef std::tuple<
...
+    void set_api_entity() override { api_entity = ov::test::utils::ov_entity::ie_infer_request; }
+
     inline InferenceEngine::InferRequest createInferRequestWithConfig() {
         // Load config
         configuration.insert({CONFIG_KEY(EXCLUSIVE_ASYNC_REQUESTS), CONFIG_VALUE(YES)});
-        if (targetDevice.find(CommonTestUtils::DEVICE_AUTO) == std::string::npos &&
-            targetDevice.find(CommonTestUtils::DEVICE_MULTI) == std::string::npos &&
-            targetDevice.find(CommonTestUtils::DEVICE_HETERO) == std::string::npos) {
-            ie->SetConfig(configuration, targetDevice);
+        if (target_device.find(CommonTestUtils::DEVICE_AUTO) == std::string::npos &&
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/behavior/infer_request/config.hpp:60]: +batch

File: src/tests/functional/plugin/shared/include/behavior/infer_request/io_blob.hpp [hunk 3]
```diff
@@ -414,6 +416,14 @@ class InferRequestIOBBlobSetLayoutTest : public testing::WithParamInterface<Infe
         if (!configuration.empty()) {
             PluginCache::get().reset();
         }
+        auto &apiSummary = ov::test::utils::ApiSummary::getInstance();
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/behavior/infer_request/io_blob.hpp:419]: APIBaseTest::TearDown

File: src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_graph_info.hpp [hunk 4]
```diff
@@ -48,27 +48,30 @@ class OVExecGraphImportExportTest : public testing::WithParamInterface<OVExecGra
...
     ov::AnyMap configuration;
     ov::element::Type_t elementType;
     std::shared_ptr<ov::Model> function;
+
+    void set_api_entity() override { api_entity = ov::test::utils::ov_entity::ov_compiled_model; }
 };
 
 TEST_P(OVExecGraphImportExportTest, importExportedFunction) {
-    if (targetDevice == "MULTI" || targetDevice == "AUTO") {
+    if (target_device == "MULTI" || target_device == "AUTO") {
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/behavior/ov_executable_network/exec_graph_info.hpp:74]: CommonTestUtils::DEVICE_MULTI

File: src/tests/functional/plugin/shared/include/behavior/ov_plugin/properties_tests.hpp [hunk 5]
```diff
@@ -30,6 +30,7 @@ class OVEmptyPropertiesTests : public testing::WithParamInterface<std::string>,
     static std::string getTestCaseName(testing::TestParamInfo<std::string> obj);
 
     void SetUp() override;
+    void  TearDown() override;
```
Comment (iefode) [inline src/tests/functional/plugin/shared/include/behavior/ov_plugin/properties_tests.hpp:33]: Remove?

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/inference_chaining.cpp [hunk 6]
```diff
@@ -134,6 +134,20 @@ void OVInferenceChaining::Run() {
     }
 }
 
+void OVInferenceChaining::TearDown() {
+    if (!configuration.empty()) {
```
Comment (iefode) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/inference_chaining.cpp:138]: Remove?

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/plugin/core_integration.cpp [hunk 7]
```diff
@@ -69,7 +69,20 @@ INSTANTIATE_TEST_SUITE_P(
 // IE Class SetConfig
 //
 
-using IEClassSetConfigTestHETERO = BehaviorTestsUtils::IEClassNetworkTest;
+class IEClassSetConfigTestHETERO : public BehaviorTestsUtils::IEClassNetworkTest,
+                                   public virtual ov::test::behavior::APIBaseTest {
+protected:
+    void set_api_entity() override {
```
[the remark this answers is not in the document]
Comment (iefode) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/plugin/core_integration.cpp:75]: Fixed 👍

File: src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp [hunk 8]
```diff
@@ -75,7 +75,7 @@ inline const std::string generate_complex_device_name(const std::string& deviceN
     return deviceName + ":" + ov::test::conformance::targetDevice;
 }
 
-inline const std::vector<std::string> return_all_possible_device_combination() {
+static const std::vector<std::string> return_all_possible_device_combination() {
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp:78]: Had some idea, decide not to do it, forgot to revert changes :D

File: src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/main.cpp [hunk 9]
```diff
@@ -75,25 +79,42 @@ int main(int argc, char* argv[]) {
...
 
     ::testing::InitGoogleTest(&argc, argv);
-    ::testing::AddGlobalTestEnvironment(new LayerTestsUtils::TestEnvironment);
+    ::testing::AddGlobalTestEnvironment(new ov::test::utils::TestEnvironment);
 
     auto exernalSignalHandler = [](int errCode) {
         std::cerr << "Unexpected application crash with code: " << errCode << std::endl;
 
         // set default handler for crash
+        signal(SIGABRT, SIG_DFL);
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/main.cpp:93]: to handle more crashes to save reports

File: src/tests/functional/plugin/shared/include/base/behavior_test_utils.hpp [hunk 10]
```diff
@@ -77,27 +78,30 @@ class InferRequestTests : public testing::WithParamInterface<InferRequestParams>
     }
 
     void SetUp() override {
+        std::tie(target_device, configuration) = this->GetParam();
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/tests/functional/plugin/shared/include/base/behavior_test_utils.hpp:61]: we should initiate parameters to get correct report

File: src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/ov_plugin/properties.cpp [hunk 11]
```diff
@@ -62,28 +43,10 @@ const std::vector<ov::AnyMap> auto_batch_properties = {
 
 INSTANTIATE_TEST_SUITE_P(ov_plugin, OVPropertiesTests,
         ::testing::Combine(
-                ::testing::Values(ov::test::conformance::targetDevice),
+                ::testing::ValuesIn(return_all_possible_device_combination()),
                 ::testing::ValuesIn(default_properties)),
         OVPropertiesTests::getTestCaseName);
 
-INSTANTIATE_TEST_SUITE_P(ov_plugin_Hetero, OVPropertiesTests,
```
Comment (ilyachur) [inline src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/ov_plugin/properties.cpp:69]: Why did you reduce the number of tests?

File: src/tests/functional/plugin/gna/shared_tests_instances/behavior/ov_executable_network/get_metric.cpp [hunk 12]
```diff
@@ -188,19 +188,19 @@ INSTANTIATE_TEST_SUITE_P(
...
 // TODO: verify hetero interop
 INSTANTIATE_TEST_SUITE_P(DISABLED_smoke_OVClassHeteroExecutableNetworlGetMetricTest,
         OVClassHeteroExecutableNetworkGetMetricTest_NETWORK_NAME,
-        ::testing::Values("GNA"));
+        ::testing::Values("HETERO:GNA"));
 
 INSTANTIATE_TEST_SUITE_P(smoke_OVClassHeteroExecutableNetworlGetMetricTest,
         OVClassHeteroExecutableNetworkGetMetricTest_TARGET_FALLBACK,
-        ::testing::Values("GNA"));
+        ::testing::Values("HETERO:GNA"));
```
Comment (ilyachur) [inline src/tests/functional/plugin/gna/shared_tests_instances/behavior/ov_executable_network/get_metric.cpp:205]: Why do you change this test for GNA?

Comment (iefode) [reply src/tests/functional/plugin/gna/shared_tests_instances/behavior/ov_executable_network/get_metric.cpp:205]: You requested me to change the `target_device` string formation in the test

File: src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp [hunk 13]
```diff
@@ -85,8 +85,28 @@ inline const std::vector<std::string> return_all_possible_device_combination() {
     return res;
 }
 
-const std::vector<std::map<std::string, std::string>> empty_config = {
-        {},
+inline std::vector<std::pair<std::string, std::string>> generate_pairs_plugin_name_by_device() {
+    std::vector<std::pair<std::string, std::string>> res;
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp:89]: I have tried to use `map`
Gtest works with vectors only (when goes over parameters). Can't

File: src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/ov_plugin/properties.cpp (hunk 11 above)
Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/ov_plugin/properties.cpp:69]: I didn't reduce. I just replaced instantiations by `return_all_possible_devices`

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/plugin/core_integration.cpp [hunk 14]
```diff
@@ -69,7 +69,20 @@ INSTANTIATE_TEST_SUITE_P(
...
+        api_entity = ov::test::utils::ov_entity::ie_plugin;
+    }
+
+public:
+    void SetUp() override {
+        target_device = CommonTestUtils::DEVICE_HETERO;
```
[the remark this answers is not in the document]
Comment (iefode) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/plugin/core_integration.cpp:76]: Fixed

File: src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp [hunk 15]
```diff
@@ -85,8 +85,28 @@ inline const std::vector<std::string> return_all_possible_device_combination() {
     return res;
 }
 
-const std::vector<std::map<std::string, std::string>> empty_config = {
-        {},
+inline std::vector<std::pair<std::string, std::string>> generate_all_pairs() {
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/include/api_conformance_helpers.hpp]: renamed and commented

File: src/tests/functional/plugin/gna/shared_tests_instances/behavior/ov_executable_network/get_metric.cpp (hunk 12 above)
Comment (iefode) [reply src/tests/functional/plugin/gna/shared_tests_instances/behavior/ov_executable_network/get_metric.cpp]: reverted

File: src/tests/functional/plugin/shared/include/base/behavior_test_utils.hpp [hunk 16]
```diff
@@ -130,13 +112,70 @@ class IEClassNetworkTest : public ov::test::behavior::OVClassNetworkTest {
     }
 };
 
-class IEClassBaseTestP : public IEClassNetworkTest, public ::testing::WithParamInterface<std::string> {
+// TODO
```
Comment (ilyachur) [inline src/tests/functional/plugin/shared/include/base/behavior_test_utils.hpp:115]: TODO what?

**Review by ilyachur — APPROVED** [review]

File: src/tests/functional/plugin/shared/include/base/behavior_test_utils.hpp (hunk 16 above)
Comment (iefode) [reply src/tests/functional/plugin/shared/include/base/behavior_test_utils.hpp]: Removed, thanks :)

**Pull request merged**